src/equations/polytropic_euler_2d.jl

```diff
@@ -0,0 +1,255 @@
+    rho, v1, v2 = cons2prim(u, equations)
+
+    # Residual from Winters (2019) [0.1007/s10543-019-00789-w] eq. (5.2).
+    h = 8 + cos(2 * pi * x[1]) * sin(2 * pi * x[2]) * cos(2 * pi * t)
+    h_t = -2 * pi * cos(2 * pi * x[1]) * sin(2 * pi * x[2]) * sin(2 * pi * t)
+    h_x = -2 * pi * sin(2 * pi * x[1]) * sin(2 * pi * x[2]) * cos(2 * pi * t)
+    h_y = 2 * pi * cos(2 * pi * x[1]) * cos(2 * pi * x[2]) * cos(2 * pi * t)
+
+    rho_x = h_x
+    rho_y = h_y
+
+    b = equations.kappa * equations.gamma * h^(equations.gamma - 1)
+
+    r_1 = h_t + h_x / 2 + 3 / 2 * h_y
+    r_2 = h_t / 2 + h_x / 4 + b * rho_x + 3 / 4 * h_y
+    r_3 = 3 / 2 * h_t + 3 / 4 * h_x + 9 / 4 * h_y + b * rho_y
+
+    return SVector(r_1, r_2, r_3)
+end
+
+"""
+    initial_condition_weak_blast_wave(x, t, equations::PolytropicEulerEquations2D)
+
+A weak blast wave adapted from
+- Sebastian Hennemann, Gregor J. Gassner (2020)
+  A provably entropy stable subcell shock capturing approach for high order split form DG
+  [arXiv: 2008.12044](https://arxiv.org/abs/2008.12044)
+"""
+function initial_condition_weak_blast_wave(x, t, equations::PolytropicEulerEquations2D)
+    # Adapted MHD version of the weak blast wave from Hennemann & Gassner JCP paper 2020 (Sec. 6.3)
+    # Set up polar coordinates
+    inicenter = (0, 0)
+    x_norm = x[1] - inicenter[1]
+    y_norm = x[2] - inicenter[2]
+    r = sqrt(x_norm^2 + y_norm^2)
+    phi = atan(y_norm, x_norm)
+
+    # Calculate primitive variables
+    rho = r > 0.5 ? 1.0 : 1.1691
+    v1 = r > 0.5 ? 0.0 : 0.1882 * cos(phi)
+    v2 = r > 0.5 ? 0.0 : 0.1882 * sin(phi)
+
+    return prim2cons(SVector(rho, v1, v2), equations)
+end
+
+# Calculate 1D flux for a single point in the normal direction
+# Note, this directional vector is not normalized
+@inline function flux(u, normal_direction::AbstractVector,
+                      equations::PolytropicEulerEquations2D)
+    rho, v1, v2 = cons2prim(u, equations)
+    p = pressure(u, equations)
+
+    v_normal = v1 * normal_direction[1] + v2 * normal_direction[2]
+    rho_v_normal = rho * v_normal
+    f1 = rho_v_normal
+    f2 = rho_v_normal * v1 + p * normal_direction[1]
+    f3 = rho_v_normal * v2 + p * normal_direction[2]
+    return SVector(f1, f2, f3)
+end
+
+"""
+flux_winters_etal(u_ll, u_rr, normal_direction,
+                  equations::PolytropicEulerEquations2D)
+
+Entropy conserving two-point flux for isothermal or polytropic gases.
+Requires a special weighted Stolarsky mean for the evaluation of the density
+denoted here as `stolarsky_mean`. Note, for isothermal gases where `gamma = 1`
+this `stolarsky_mean` becomes the [`ln_mean`](@ref).
+
+For details see Section 3.2 of the following reference
+- Andrew R. Winters, Christof Czernik, Moritz B. Schily & Gregor J. Gassner (2020)
+Entropy stable numerical approximations for the isothermal and polytropic
+Euler equations
+[DOI: 10.1007/s10543-019-00789-w](https://doi.org/10.1007/s10543-019-00789-w)
```

Comment:
```suggestion
- Andrew R. Winters, Christof Czernik, Moritz B. Schily & Gregor J. Gassner (2020)
  Entropy stable numerical approximations for the isothermal and polytropic
  Euler equations
  [DOI: 10.1007/s10543-019-00789-w](https://doi.org/10.1007/s10543-019-00789-w)
```
Similar here, one must indent the reference with two spaces to format correctly in the docs.